action/protocol/rewarding/reward.go

```diff
@@ -215,7 +215,7 @@ func (p *Protocol) GrantEpochReward(
 	}
 
 	// Reward additional bootstrap bonus
-	if epochNum <= a.foundationBonusLastEpoch {
+	if epochNum <= a.foundationBonusLastEpoch || (epochNum >= a.foundationBonusP2StartEpoch && epochNum <= a.foundationBonusP2EndEpoch) {
```

Comment:
somewhere in reward protocol, we need to do
if epoch is Fairbank {
    a.foundationBonusP2StartEpoch = genesis.xxx
    a.foundationBonusP2EndEpoch = genesis.yyy
    p.state(sm, adminKey, &a)
}
that is to write the 2 new values at Fairbank height

then this logic can follow

Reply:
or better off
if epoch == Fairbank - 1
to write it in advance

then at Fairbank, this is ready to hit